rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,499 @@
+
+### Impossible to differentiate from internal comments
+
+The lack of a formal definition of a doc-comment also means there is no reliable way to distinguish them from internal comments, which makes it impossible to access from doc-comments from tooling. Futhermore it makes it very hard to generate accurate and complete reference documentation.
+
+### References to the problems above
+
+This curated link collection highlights the Nix ecosystem's inconsistencies, a primary focus of this RFC.
+
+#### nixpkgs - comment examples
+
+- [lib/attrsets](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/lib/attrsets.nix)
+- [trivial-builders](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/pkgs/build-support/trivial-builders/default.nix)
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/nixos/lib/make-disk-image.nix)
+
+# Design
+[design]: #detailed-design
+
+In the following we give a comprehensive overview to our decision that we've made. Detailed arguments for and against every decision can be found in the [Decisions](#Decisions) section
+
+## CommonMark
+
+**CommonMark according to [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md) is the content of all doc-comments.**
+
+Adopting CommonMark as the content for all doc-comments brings the benefits of widely accepted and understood documentation format in tech projects, while maintaining profitability and consistency within the Nix ecosystem by aligning with existing [NixOS/RFC-72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).
+
+## `/** */` is the doc-comment format
+
+The decision to use /** to start a doc-comment ensures a unique distinction from regular comments while still allowing seamless writing without IDE or editor support. A single choice provides the best developer experience and simplifies tooling.
+
+## Placement
+
+**The placement describes the relationship between doc-comments and the documentable node.**
+
+A **documentable node** can be:
+
+- Expression 
+- Binding 
+- Lambda Formal
+
+The following rules apply in descending order of precedence:
+
+- Doc-comments are placed before the **documentable node**. Only whitespace is allowed in between. ([Examples](#basic-examples)) 
+
+- The documentation present before the `attribute path` describes the body of the attribute. ([Examples](#Attributes))
+    - In case placement is ambiguous, the one closer to the body has higher precedence. ([Examples](#ambiguous-placement))
+         
+- All partial functions of a curried lambda can share the same placement with the outermost lambda. ([Examples](#partial-lambda-functions))
```

Comment:
Nitpick:

```suggestion
- All partial applications of a curried lambda can share the same placement with the outermost lambda. ([Examples](#partial-lambda-functions))
```

And elsewhere.

"Partial function" means a function which might not terminate (i.e. might go into an infinite loop).  What you're describing here is a *partially applied* function.